src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -256,6 +254,32 @@ public Instance(
         }
     }
 
+    /** this function is unsafe, please call holding a lock. */
+    public synchronized void updateServiceList(List<String> services) {
```

Comment:
Instead of having a method where we pass the list of services, what do you think about adding a constructor parameter that is an instance of an interface like:

```
interface ServiceNameProvider {
  List<String> getServiceNames();
}
```
Then we can own that object and you can create a simple one for the basic case